streams/src/test/java/org/apache/kafka/streams/integration/PurgeRepartitionTopicIntegrationTest.java

```diff
@@ -0,0 +1,223 @@
+        streamsConfiguration.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.Integer().getClass());
+        streamsConfiguration.put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory(APPLICATION_ID).getPath());
+        streamsConfiguration.put(StreamsConfig.topicPrefix(TopicConfig.SEGMENT_MS_CONFIG), purgeIntervalMs);
+        streamsConfiguration.put(StreamsConfig.topicPrefix(TopicConfig.SEGMENT_BYTES_CONFIG), purgeSegmentBytes);
+        streamsConfiguration.put(StreamsConfig.producerPrefix(ProducerConfig.BATCH_SIZE_CONFIG), purgeSegmentBytes / 2);    // we cannot allow batch size larger than segment size
+        streamsConfiguration.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, purgeIntervalMs);
+
+        StreamsBuilder builder = new StreamsBuilder();
+
+        builder.stream(INPUT_TOPIC)
+               .groupBy(MockKeyValueMapper.SelectKeyKeyValueMapper())
+               .count();
+
+        System.out.println(builder.build().describe());
+
+        kafkaStreams = new KafkaStreams(builder.build(), new StreamsConfig(streamsConfiguration), time);
+    }
+
+    @After
+    public void shutdown() throws IOException {
+        if (kafkaStreams != null) {
+            kafkaStreams.close(30, TimeUnit.SECONDS);
+        }
+    }
+
+
+    @Test
+    public void shouldRestoreState() throws InterruptedException {
+        // produce some data to input topic
+        final Properties producerConfig = new Properties();
+        producerConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+
+        try (final KafkaProducer<Integer, Integer> producer =
+                     new KafkaProducer<>(producerConfig, new IntegerSerializer(), new IntegerSerializer())) {
+
+            for (int i = 0; i < 1000; i++) {
+                producer.send(new ProducerRecord<>(INPUT_TOPIC, i, i));
+            }
+        }
+
+        kafkaStreams.start();
+
+        TestUtils.waitForCondition(new RepartitionTopicCreated(), 60000,
+                "Repartition topic " + REPARTITION_TOPIC + " not created with the expected configs after 60000 ms.");
+
+        TestUtils.waitForCondition(
+                new RepartitionTopicPurged(new TopicSizeVerifier() {
+                    @Override
+                    public boolean verify(long currentSize) {
+                        return currentSize > 0;
+                    }
+                }),
+                60000,
+                "Repartition topic " + REPARTITION_TOPIC + " not received data after 60000 ms."
+        );
+
+        // we need long enough timeout to by-pass the log manager's InitialTaskDelayMs, which is hard-coded on server side
+        TestUtils.waitForCondition(
+                new RepartitionTopicPurged(new TopicSizeVerifier() {
+                    @Override
+                    public boolean verify(long currentSize) {
+                        return currentSize <= purgeSegmentBytes;
```

Comment:
Must it not be `<` ? If the latest segment is full, won't we roll it and thus it would get purged?

Reply:
We will roll if the segment size `>` segment bytes, not `>=`.